src/Core/Content/Product/SalesChannel/Listing/Processor/SortingListingProcessor.php

```diff
@@ -91,34 +90,25 @@ private function getCurrentSorting(ProductSortingCollection $sortings, Request $
 
     private function getAvailableSortings(Request $request, Context $context): ProductSortingCollection
     {
-        $criteria = new Criteria();
-        $criteria->setTitle('product-listing::load-sortings');
-        /** @var string[] $availableSortings */
-        $availableSortings = $request->get('availableSortings');
-        $availableSortingsById = [];
+        /** @var ProductSortingCollection $availableSortings */
+        $availableSortings = $request->get('availableSortingCollection');
 
         if ($availableSortings) {
-            arsort($availableSortings, \SORT_DESC | \SORT_NUMERIC);
-            $availableSortingsFilter = array_keys($availableSortings);
-
-            $availableSortingsById = array_filter($availableSortingsFilter, fn ($filter) => Uuid::isValid($filter));
+            $sortings = new ProductSortingCollection();
+            foreach ($availableSortings as $availableSorting) {
+                $sortings->add($availableSorting);
+            }
 
-            $filter = new EqualsAnyFilter('id', $availableSortingsById);
-
-            $criteria->addFilter($filter);
+            return $sortings;
         }
```

Comment:
you do not respect the old parameter `$request->get('availableSortings');` anymore. as the routes can also be called directly, this would cause a break

Reply:
It was only used for custom sorting; yes, the naming was confusing. We now do the same with the new `restrictedProductSortingCollection`.